### PR TITLE
Dc 156 add database entities

### PR DESCRIPTION
**Pull request by LavMatt**
This PR relates to https://github.com/ministryofjustice/find-moj-data/issues/156

Front end now recognises Databases (datahub container entities). The changes made should still work with the dev and test datahub instances where we have the metadata ingested classified different and a different set of domains.

It makes use of `ministryofjustice-data-platform-catalogue v0.22.0` which builds in additional entities and methods to get the parents and children of datasets (both data products and containers) and containers.

### CODE REVIEW

File: templates/details_dataset.html
[the remark this answers is not in the document]
Comment (LavMatt) [reply templates/details_dataset.html:1]: Quite possibly yeah. Database is a subtype of container (and that's the name used for that template) so we should probably align and do the same for both entity types Dataset/Table Container/Database

File: home/service/details.py [hunk 1]
```diff
@@ -56,40 +57,94 @@ def _get_context(self):
...
+
+        entities_in_database = sorted(entities_in_database, key=lambda d: d["name"])
+
+        return entities_in_database
+
+    def _get_context(self):
+        context = {
+            "result": self.result,
+            "result_type": (
+                "Database"
```
Comment (murdo-moj) [inline home/service/details.py:101]: isn't the result_type always database here?

File: home/service/search.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def domains_with_their_subdomains(domain: str, subdomain: str) -> list[str]:
 
     subdomains = DomainModel().subdomains.get(domain, [])
     subdomains = [subdomain[0] for subdomain in subdomains]
-    return [domain, *subdomains] if subdomains else []
+    return [domain, *subdomains] if not domain == "" else []
```
Comment (murdo-moj) [inline home/service/search.py:28]:
```suggestion
    return [domain, *subdomains] if domain != "" else []
```

File: templates/details_database.html [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  <div class="govuk-grid-column-one-third">
+    {% include "partial/contact_info.html" with data_owner=result.metadata.owner data_owner_email=result.metadata.owner_email %}
+  </div>
+</div>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <table class="govuk-table">
+      <caption class="govuk-table__caption govuk-table__caption--m">Data product content</caption>
```
Comment (murdo-moj) [inline templates/details_database.html:74]:
```suggestion
      <caption class="govuk-table__caption govuk-table__caption--m">Database content</caption>
```

File: home/service/details.py (hunk 1 above)
Comment (LavMatt) [reply home/service/details.py:101]: Yeah i think it's not needed and was accommodating dodgy test inputs rather than actually being of any use.

Comment (LavMatt) [reply home/service/details.py]: have committed fix

**Review by murdo-moj — APPROVED** [review]

**Pull request merged**